@here/olp-sdk-dataservice-read/test/unit/CatalogClient.test.ts

```diff
@@ -149,6 +149,36 @@ describe("CatalogClient", () => {
             });
     });
 
+    it("Should method getLayerVersions return HttpError when MetadataApi.getLayerVersions crashes", async () => {
+        const TEST_ERROR_CODE = 404;
+        const mockedError = new HttpError(
+            TEST_ERROR_CODE,
+            "Can not get catalog layer version"
+        );
+
+        getLayerVersionsStub.callsFake(
+            (builder: any, params: any): Promise<MetadataApi.LayerVersions> => {
+                return Promise.reject(mockedError);
```

Comment:
You could use simple reject like so
`return Promise.reject({
  status: 404,
  statusText: "Can not get catalog layer version"
});`

Reply:
fixed